lib/preprocess/ConfigPreprocessorOverride.ts

```diff
@@ -0,0 +1,111 @@
+import type { Resource } from 'rdf-object';
+import type { RdfObjectLoader } from 'rdf-object/lib/RdfObjectLoader';
+import type { Logger } from 'winston';
+import type { IConfigPreprocessor, IConfigPreprocessorTransform } from './IConfigPreprocessor';
+
+// TODO:
+export class ConfigPreprocessorOverride implements IConfigPreprocessor<Record<string, Resource>> {
+  public readonly objectLoader: RdfObjectLoader;
+  public readonly componentResources: Record<string, Resource>;
+
+  private overrides: Record<string, Record<string, Resource>> | undefined;
+
+  public constructor(options: IComponentConfigPreprocessorOverrideOptions) {
+    this.objectLoader = options.objectLoader;
+    this.componentResources = options.componentResources;
+  }
+
+  public canHandle(config: Resource): Record<string, Resource> | undefined {
+    if (!this.overrides) {
+      this.overrides = this.findOverrides();
+    }
+    return this.overrides[config.value];
+  }
+
+  public transform(config: Resource, handleResponse: Record<string, Resource>): IConfigPreprocessorTransform {
+    for (const id of Object.keys(config.properties)) {
+      const overrideValue = handleResponse[id];
+      if (overrideValue) {
+        config.properties[id] = [ overrideValue ];
+      }
+    }
+    return { rawConfig: config, finished: false };
+  }
+
+  // TODO: errors + logging
+  // TODO: split up
+  public findOverrides(): Record<string, Record<string, Resource>> {
+    const overrideUri = this.objectLoader.contextResolved.expandTerm('oo:Override')!;
+    const overrideInstanceUri = this.objectLoader.contextResolved.expandTerm('oo:overrideInstance')!;
+    const overrideChains: Record<string, Resource[]> = {};
+    for (const [ id, resource ] of Object.entries(this.objectLoader.resources)) {
+      if (resource.isA(overrideUri) && resource.value !== overrideUri) {
+        const targets = resource.properties[overrideInstanceUri];
+        if (!targets || targets.length !== 1) {
+          throw new Error(`${id} needs exactly 1 overrideInstance`);
```

Comment:
Should probably become `ErrorResourcesContext`, so that we can include the resource.